sorc/ncep_post.fd/SURFCE.f

```diff
@@ -1220,6 +1221,229 @@ SUBROUTINE SURFCE
+         IF (ID(18)<0) ID(18) = 0
+        if(grib=='grib2') then
+          cfld=cfld+1
+           fld_info(cfld)%ifld=IAVBLFLD(IGET(999))
+           fld_info(cfld)%ntrange=1
+          fld_info(cfld)%tinvstat=IFHR-ID(18)
+!$omp parallel do private(i,j,jj)
+              do j=1,jend-jsta+1
+                jj = jsta+j-1
+                do i=1,im
+                  datapd(i,j,cfld) = GRID1(i,jj)
+                enddo
+              enddo
+            endif
+          ENDIF
+
+          IF ( IGET(1000)>0 )THEN
+!$omp parallel do private(i,j)
+         DO J=JSTA,JEND
+           DO I=1,IM
+             GRID1(I,J) = TETRAN(I,J)
+           ENDDO
+         ENDDO
+         ID(1:25) = 0
+         ITPREC     = NINT(TPREC)
+!mp
+         if (ITPREC /= 0) then
+           IFINCR     = MOD(IFHR,ITPREC)
+           IF(IFMIN >= 1)IFINCR= MOD(IFHR*60+IFMIN,ITPREC*60)
+         else
+           IFINCR     = 0
+         endif
+!mp
+         ID(18)     = 0
+         ID(19)     = IFHR
+         IF(IFMIN >= 1)ID(19)=IFHR*60+IFMIN
+         ID(20)     = 4
+         IF (IFINCR==0) THEN
+           ID(18) = IFHR-ITPREC
+         ELSE
+           ID(18) = IFHR-IFINCR
+           IF(IFMIN >= 1)ID(18)=IFHR*60+IFMIN-IFINCR
+         ENDIF
+         IF (ID(18)<0) ID(18) = 0
+        if(grib=='grib2') then
+          cfld=cfld+1
+           fld_info(cfld)%ifld=IAVBLFLD(IGET(1000))
+           fld_info(cfld)%ntrange=1
+          fld_info(cfld)%tinvstat=IFHR-ID(18)
+!$omp parallel do private(i,j,jj)
+          do j=1,jend-jsta+1
+            jj = jsta+j-1
+            do i=1,im
+              datapd(i,j,cfld) = GRID1(i,jj)
+            enddo
+          enddo
+        endif
+      ENDIF
+!
+          IF ( IGET(1001)>0 )THEN
+!$omp parallel do private(i,j)
+         DO J=JSTA,JEND
+           DO I=1,IM
+             GRID1(I,J) = TEDIR(I,J)
+           ENDDO
+         ENDDO
+         ID(1:25) = 0
+         ITPREC     = NINT(TPREC)
+!mp
+         if (ITPREC /= 0) then
+           IFINCR     = MOD(IFHR,ITPREC)
+           IF(IFMIN >= 1)IFINCR= MOD(IFHR*60+IFMIN,ITPREC*60)
+         else
+           IFINCR     = 0
+         endif
+!mp
+         ID(18)     = 0
+         ID(19)     = IFHR
+         IF(IFMIN >= 1)ID(19)=IFHR*60+IFMIN
+         ID(20)     = 4
+         IF (IFINCR==0) THEN
+           ID(18) = IFHR-ITPREC
+         ELSE
+           ID(18) = IFHR-IFINCR
+           IF(IFMIN >= 1)ID(18)=IFHR*60+IFMIN-IFINCR
+         ENDIF
+         IF (ID(18)<0) ID(18) = 0
+        if(grib=='grib2') then
+          cfld=cfld+1
+           fld_info(cfld)%ifld=IAVBLFLD(IGET(1001))
+           fld_info(cfld)%ntrange=1
+          fld_info(cfld)%tinvstat=IFHR-ID(18)
+!$omp parallel do private(i,j,jj)
+          do j=1,jend-jsta+1
+            jj = jsta+j-1
+            do i=1,im
+              datapd(i,j,cfld) = GRID1(i,jj)
+            enddo
+          enddo
+        endif
+      ENDIF
+!
+
+         IF (IGET(1002)>0) THEN
+          IF(MODELNAME == 'NCAR'.OR.MODELNAME=='RSM' .OR. &
```

Comment:
@HelinWei-NOAA I don't think a specified setting is needed for RAPR, NCAR, RSM models since this is total new field.